src/network/networkd-route.c

```diff
@@ -617,6 +579,12 @@ int route_configure(
         if (r < 0)
                 return log_error_errno(r, "Could not append RTA_PREF attribute: %m");
 
+        if (route->lifetime != SEC_INFINITY) {
+                r = sd_netlink_message_append_u32(req, RTA_EXPIRES, route->lifetime);
+                if (r < 0)
```

Comment:
When this fails, then we should go to fallback logic, that is using `sd_event_add_time`. What do you think? This way, we can safely support old kernels.

Reply:
FTR, since github is attaching an old comment here: this won't work, because we don't communicate with the kernel yet.